home/private_dot_local/lib/agent-equipment/agent_equipment/validator.py

```
@@ -1823,7 +1854,70 @@ def _catalog_distribution_is_valid(distribution: Any) -> bool:
+            {"kind", "repository"},
+            {"branch", "kind", "repository"},
+        ):
+            return False
+        repository = source.get("repository")
+        return (
+            _source_string_is_bounded(repository)
+            and _public_git_repository_is_valid(repository)
+            and (
+                "branch" not in source
+                or isinstance(source.get("branch"), str)
+                and _git_branch_is_valid(source["branch"])
```

Ruff (0.16.1)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)